downloader/src/main/kotlin/vcs/GitWorkingTree.kt

```diff
@@ -57,7 +57,7 @@ open class GitWorkingTree(
 ) : WorkingTree(workingDir, vcsType) {
     companion object : Logging
 
-    private val repo: Repository
+    val repo: Repository
```

Comment:
In the PR which introduced `repo.use` I asked whether it makes sense to extract it to a function.
Now, as you make the property public I think that would indeed make sense. So, instead of exposing `repo` expose a function to run something using `repo`. This would ensure that the contract (to use `use`) does not get violated.

Reply:
> I asked whether it makes sense to extract it to a function.
> Now, as you make the property public I think that would indeed make sense.

How would you envision such a function to look like? Should it take a lambda that is applied inside `repo.use`?

Reply:
> How would you envision such a function to look like? Should it take a lambda that is applied inside `repo.use`?

Yes, that was what I had in mind. Does it make sense?

Reply:
Yes, I agree that makes sense now. I've implemented it in a new commit.